nsd/check.py

```diff
@@ -0,0 +1,40 @@
+# stdlib
+import os
+import re
+
+# project
+from checks import AgentCheck
+from config import _is_affirmative
+from util import Platform
+from utils.subprocess_output import get_subprocess_output
+
+class NsdCheck(AgentCheck):
+    # Stats info https://www.nlnetlabs.nl/projects/nsd/nsd-control.8.html
+
+    NSD_CTL = ['nsd-control', 'stats']
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances=instances)
+        if instances is not None and len(instances) > 1:
+            raise Exception("NSD check only supports one configured instance.")
+
+    def check(self, instance):
+        if instance is None:
+            instance = {}
+
+        host = instance.get('host')
+        tags = instance.get('tags', [])
```

Comment:
We should add the host to the `tags` list. It would help splitting metrics by instances (ie previous comment).

Also `tags` should be int the `conf.yaml.example`. For example: https://github.com/DataDog/integrations-extras/blob/master/snmpwalk/conf.yaml.example#L18